src/test/java/com/example/board/service/UserServiceTest.java

```diff
@@ -0,0 +1,114 @@
+package com.example.board.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+import com.example.board.domain.auth.JWTTokenProvider;
+import com.example.board.domain.user.User;
+import com.example.board.repository.UserRepository;
+import com.example.board.request.SignInRequest;
+import com.example.board.request.SignUpRequest;
+import com.example.board.response.SignInResponse;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    @Mock
+    private JWTTokenProvider jwtTokenProvider;
+
+    @Test
+    @DisplayName("회원가입을 할 수 있다.")
+    void register() {
+        // given
+        SignUpRequest request = signUpRequest();
+        User user = createUserEntity(request);
+
+        // mocking
+        given(userRepository.save(any())).willReturn(user);
+        given(userRepository.findById(1L)).willReturn(Optional.of(user));
+
+        // when
+        Long newUserId = userService.signUp(request);
+
+        // then
+        User findUser = userRepository.findById(newUserId).get();
+
+        assertThat(user.getId()).isEqualTo(findUser.getId());
```

Comment:
```suggestion
        assertThat(user).isEqualTo(findUser);
```

equauls 를 작성하여 Entity를 직접 비교하는건 어떨까요